### PR TITLE
Close leak in stecil-setup.chpl test

### PR DESCRIPTION
**Pull request by bradcray**
I took the approach of using a cast to unmanaged to free a borrow that I
knew was of an unmanaged class, and also freed another class in the
obvious way.

### CODE REVIEW

**Comment by bradcray** [comment]
@mppf: this is the last leak case I was working on in the background during the L&L.  I was worried that more principled changes might violate the original intention of this test, so took this somewhat scrappy approach.  Thoughts?

File: test/classes/delete-free/lifetimes/stecil-setup.chpl [hunk 1]
```diff
@@ -8,22 +8,29 @@ class MyLocDom {
...
-  else
+  } else
     elem.x += 1;
 }
 
 proc test() {
   var dom = new unmanaged MyStencilDom();
   dom.setup();
   writeln(dom);
+  delete dom;
```
Comment (mppf) [inline test/classes/delete-free/lifetimes/stecil-setup.chpl]: If it's just for this test, I think it makes more sense to just delete dom.A[1] here just before the delete dom, but I'm also OK with what you wrote above (I just worry that it being in the deinit makes it look more general purpose than it is).

Comment (bradcray) [reply test/classes/delete-free/lifetimes/stecil-setup.chpl]: So you're suggesting using `delete _to_unmanaged(dom.A[1]);` just above this line, is that right?

Comment (mppf) [reply test/classes/delete-free/lifetimes/stecil-setup.chpl]: Yes, exactly. BTW you should be able to write `delete dom.A[1]:unmanaged` (although parens might be required there somewhere).

Comment (bradcray) [reply test/classes/delete-free/lifetimes/stecil-setup.chpl]: Oh, weird...  I'd originally tried casting to `unmanaged MyLocDom` to delete it, but that didn't work:

```
stecil-setup.chpl:27: error: illegal cast from MyLocDom to unmanaged MyLocDom
```

and it didn't occur to me to try just `: unmanaged`.  Is the error message from my attempt a bug or a feature?

Comment (mppf) [reply test/classes/delete-free/lifetimes/stecil-setup.chpl]: Probably a bug.

Comment (bradcray) [reply test/classes/delete-free/lifetimes/stecil-setup.chpl]: See PR #12683.  I'll also open an issue if there isn't one already.

**Pull request merged**